ci_framework/roles/edpm_prepare/tasks/main.yml

```diff
@@ -0,0 +1,98 @@
+- name: Prepare storage in CRC
+  when:
+    - cifmw_use_crc is defined
+    - cifmw_use_crc | bool
+  ansible.builtin.include_role:
+    name: 'install_yamls_makes'
+    tasks_from: 'make_crc_storage'
+
+- name: Create install_yamls output directory
+  ansible.builtin.file:
+    path: "{{ cifmw_edpm_prepare_intall_yamls_output_dir }}"
+    state: directory
+
+- name: Set install_yamls Makefile environment variables
+  ansible.builtin.set_fact:
+    cifmw_edpm_prepare_common_env: |
+      OUT: {{ cifmw_edpm_prepare_intall_yamls_output_dir }}
+    cifmw_edpm_prepare_make_openstack_env: |
+      {% if cifmw_operator_build_meta_name in cifmw_operator_build_output.operators %}
+      OPENSTACK_IMG: {{ cifmw_operator_build_output.operators[cifmw_operator_build_meta_name].image }}
+      {% endif %}
+    cifmw_edpm_prepare_make_openstack_deploy_env: |
+      {% if cifmw_operator_build_meta_name in cifmw_operator_build_output.operators %}
+      OPENSTACK_BRANCH: ""
+      GIT_CLONE_OPTS: "-l"
+      OPENSTACK_REPO: cifmw_operator_build_output.operators[cifmw_operator_build_meta_name].git_src_dir
```

Comment:
Based on my testing with [EDPM Zuul job pr ](https://github.com/openstack-k8s-operators/ci-framework/pull/154). I am hitting following[ error ](https://logserver.rdoproject.org/54/154/a95182f33941efb8c829a18b600d9fcec54a6b4d/github-check/crc-podified-edpm-deployment/e21972c/job-output.txt)
```
atal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'cifmw_operator_build_output' is undefined. 'cifmw_operator_build_output' is undefined\n\nThe error appears to be in '/home/zuul-worker/src/github.com/openstack-k8s-operators/ci-framework/ci_framework/roles/edpm_prepare/tasks/main.yml': line 14, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Set install_yamls Makefile environment variables\n  ^ here\n"}
```
cifmw_operator_build_output comes from operator_build role. But in the above job, we are not building operators that' why this var is not set during runtime.

Reply:
@pablintino  When we use make_install_yamls role, for each task we have specific vars to manipulate their environment and params for a make target.
For example in case of [make openstack_deploy
](https://logserver.rdoproject.org/58/158/e939b97efe6f02da47c3d1141d537d3e93b9a306/github-check/cifmw-end-to-end/5b0ed1e/ci-framework/artifacts/roles/install_yamls_makes/tasks/make_openstack_deploy.yml)
we have make_openstack_deploy_environment and make_openstack_deploy_params vars to do the same.

If an user defines these vars in their var files then these vars will be ignored if they use the edpm_prepare role.
Can we check if these vars for example make_openstack_deploy_params is defined, skip these kinds of tasks?

Do let me know your thoughts on this.